sqlmesh/core/engine_adapter/snowflake.py

```diff
@@ -326,7 +349,40 @@ def _get_data_objects(
         ]
 
     def set_current_catalog(self, catalog: str) -> None:
-        self.execute(exp.Use(this=exp.to_identifier(catalog)))
+        self.execute(exp.Use(this=normalize_identifiers(catalog, dialect=self.dialect)))
+
+    def set_current_schema(self, schema: str) -> None:
+        self.execute(exp.Use(kind="SCHEMA", this=to_schema(schema)))
+
+    def _to_sql(self, expression: exp.Expression, quote: bool = True, **kwargs: t.Any) -> str:
+        # note: important to use self._default_catalog instead of the self.default_catalog property
+        # otherwise we get RecursionError: maximum recursion depth exceeded
+        # because it calls get_current_catalog(), which executes a query, which needs the default catalog, which calls get_current_catalog()... etc
+        if self._default_catalog:
+            default_catalog_compare = exp.to_identifier(self._default_catalog).output_name.replace(
```

Comment:
Is there a better way to unquote things that are potentially quoted?

Reply:
exp.to_identifier(...).name, there will be no quotes